packages/parameters/src/AppConfigProvider.ts

```diff
@@ -0,0 +1,111 @@
+import { BaseProvider, DEFAULT_PROVIDERS } from 'BaseProvider';
+import {
+  AppConfigDataClient,
+  StartConfigurationSessionCommand,
+  GetLatestConfigurationCommand,
+} from '@aws-sdk/client-appconfigdata';
+import type {
+  StartConfigurationSessionCommandInput,
+  GetLatestConfigurationCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { AppConfigGetOptionsInterface } from './types/AppConfigProvider';
+
+class AppConfigProvider extends BaseProvider {
+  public client: AppConfigDataClient;
+  private application?: string;
+  private environment?: string;
+  private token: string | undefined;
+
+  /**
+   * It initializes the AppConfigProvider class with an optional set of options like region: 'us-west-1'.
+   * *
+   * @param {AppConfigDataClientConfig} options
+   */
+  public constructor(options: AppConfigGetOptionsInterface = {}) {
+    super();
+    this.client = new AppConfigDataClient(options.config || {});
+    this.environment = options.environment;
+    this.application = options.application || 'service_undefined'; // TODO: get value from ENV VARIABLES
+  }
+
+  /**
+   * Retrieve a parameter value from AWS App config.
+   *
+   * @param {string} name - Name of the configuration
+   * @param {StartConfigurationSessionCommandInput} [sdkOptions] - SDK options to propagate to `StartConfigurationSession` API call
+   * @returns {Promise<string | undefined>}
+   */
+  protected async _get(
+    name: string,
+    options?: AppConfigGetOptionsInterface
+  ): Promise<string | undefined> {
+    /**
+     * The new AppConfig APIs require two API calls to return the configuration
+     * First we start the session and after that we retrieve the configuration
+     * We need to store the token to use in the next execution
+     **/
+    if (!this.token) {
+      const sessionOptions: StartConfigurationSessionCommandInput = {
+        ConfigurationProfileIdentifier: name,
+        EnvironmentIdentifier: this.environment,
+        ApplicationIdentifier: this.application,
+      };
+      if (options && options.hasOwnProperty('sdkOptions')) {
+        Object.assign(sessionOptions, options.sdkOptions);
+      }
+
+      const sessionCommand = new StartConfigurationSessionCommand(
+        sessionOptions
+      );
+
+      const session = await this.client.send(sessionCommand);
+      this.token = session.InitialConfigurationToken;
+    }
+
+    const getConfigurationCommand = new GetLatestConfigurationCommand({
+      ConfigurationToken: this.token,
+    });
+    const response = await this.client.send(getConfigurationCommand);
+
+    //  Should we cache and flush the token after 24 hours?
+    //  NextPollConfigurationToken expires in 24 hours after the last request and causes `BadRequestException`
+    this.token = response.NextPollConfigurationToken;
+
+    const configuration = response.Configuration;
+
+    // should we convert Uint8Array to string?
```

Comment:
So, keeping in mind that this method is called by `BaseProvider.get`, I think this method should return the value as-is and leave the caller decide whether to apply transforms or not based on the arguments passed by the user.

Thoughts?

Reply:
I agree. I was confused by the `str` return type of the function. I opened an [issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/1796) in the python repo about the return type

Reply:
After implementing the SecretsProvider (in #1206) which can also return `Uint8Array` I'm convinced that we should return the value as-is and then let the user decide on whether it should be transformed or not via the `transform` parameter.

I have updated the `BaseProvider` (#1205 - merged) to support this, you might want to rebase to get the latest from `main`.

Based on the current info I think our implementation is the correct one, but I'm open to get convinced otherwise and change it back if the discussion in the issue you opened requires so.